presto-main/src/main/java/com/facebook/presto/operator/aggregation/AbstractMinMaxAggregationFunction.java

```diff
@@ -0,0 +1,325 @@
+                state.setBoolean(value);
+            }
+        }
+        catch (Throwable t) {
+            throw internalError(t);
+        }
+    }
+
+    public static void minInput(Type type, BlockPositionState state, Block block, int position)
+    {
+        if (state.getBlock() == null) {
+            state.setBlock(block);
+            state.setPosition(position);
+            return;
+        }
+        if (type.compareTo(block, position, state.getBlock(), state.getPosition()) < 0) {
+            state.setBlock(block);
+            state.setPosition(position);
+        }
+    }
+
+    public static void maxInput(Type type, BlockPositionState state, Block block, int position)
+    {
+        if (state.getBlock() == null) {
+            state.setBlock(block);
+            state.setPosition(position);
+            return;
+        }
+        if (type.compareTo(block, position, state.getBlock(), state.getPosition()) > 0) {
+            state.setBlock(block);
+            state.setPosition(position);
+        }
+    }
+
+    public static void combine(MethodHandle methodHandle, NullableLongState state, NullableLongState otherState)
+    {
+        if (state.isNull()) {
+            state.setNull(false);
+            state.setLong(otherState.getLong());
+            return;
+        }
+        try {
+            if ((boolean) methodHandle.invokeExact(otherState.getLong(), state.getLong())) {
+                state.setLong(otherState.getLong());
+            }
+        }
+        catch (Throwable t) {
+            throw internalError(t);
+        }
+    }
+
+    public static void combine(MethodHandle methodHandle, NullableDoubleState state, NullableDoubleState otherState)
+    {
+        if (state.isNull()) {
+            state.setNull(false);
+            state.setDouble(otherState.getDouble());
+            return;
+        }
+        try {
+            if ((boolean) methodHandle.invokeExact(otherState.getDouble(), state.getDouble())) {
+                state.setDouble(otherState.getDouble());
+            }
+        }
+        catch (Throwable t) {
+            throw internalError(t);
+        }
+    }
+
+    public static void combine(MethodHandle methodHandle, NullableBooleanState state, NullableBooleanState otherState)
+    {
+        if (state.isNull()) {
+            state.setNull(false);
+            state.setBoolean(otherState.getBoolean());
+            return;
+        }
+        try {
+            if ((boolean) methodHandle.invokeExact(otherState.getBoolean(), state.getBoolean())) {
+                state.setBoolean(otherState.getBoolean());
+            }
+        }
+        catch (Throwable t) {
+            throw internalError(t);
+        }
+    }
+
+    public static void minCombine(Type type, BlockPositionState state, BlockPositionState otherState)
+    {
+        if (state.getBlock() == null) {
+            state.setBlock(otherState.getBlock());
+            state.setPosition(otherState.getPosition());
```

Comment:
@highker : But it's just two lines and seems quite straightforward. Not sure if worthy it :)

Reply:
@highker  : I think this might be better: to have `BlockPostionState` manually written, and has a single method called `setBlockAndPosition`.

It doesn't make sense to set them separately.